src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSBucketCreate.java

```diff
@@ -192,11 +193,29 @@ public boolean failIfExists() {
 
     void validate(FailureCollector collector) {
       if (!containsMacro("paths")) {
+        Storage storage = null;
+        try {
+          Credentials credentials = getServiceAccountFilePath() == null ?
+            null : GCPUtils.loadServiceAccountCredentials(getServiceAccountFilePath());
+          storage = GCPUtils.getStorage(getProject(), credentials);
+        } catch (IOException e) {
+          collector.addFailure(e.getMessage(), "Ensure you entered the correct file path.")
+            .withConfigProperty(NAME_SERVICE_ACCOUNT_FILE_PATH)
+            .withStacktrace(e.getStackTrace());
+        }
         for (String path : getPaths()) {
           try {
-            GCSPath.from(path);
+            GCSPath gcsPath = GCSPath.from(path);
+            // Check if bucket exists and is accessible
+            if (storage != null) {
+              storage.get(gcsPath.getBucket());
```

Comment:
It shouldn't be a validation error if the bucket doesn't exist during the pipeline deployment time. Also, what does the getProject() returns if it is auto-detect? Does it return the project where CDAP is running or the target Dataproc project?

Reply:
I believe it should be where the getProject() is run since it just checks for the GOOGLE_CLOUD_PROJECT environment variable where it is running. I.e. CDAP during configuration, dataproc during runtime?